docs/autoprocess.py

```diff
@@ -0,0 +1,401 @@
+        :rtype: nodes.field_list
+        """
+        # Make a copy so the function parameter is not modified.
+        data = data.copy()
+
+        properties_list = nodes.field_list()
+
+        # Remove the name (it is already used in field_name...)
+        data.pop('name')
+
+        # changing the order of elements in this list affects
+        # the order in which they are displayed
+        property_names = [
+            'label',
+            'type',
+            'description',
+            'required',
+            'disabled',
+            'hidden',
+            'default',
+            'placeholder',
+            'validate_regex',
+            'choices',
+            'collapse',
+            'group']
+
+        for name in property_names:
+            if name in data:
+                value = data.pop(name)
+
+                # Value should be formatted in code-style (=literal) mode
+                if name in ['type', 'default', 'placeholder', 'validate_regex']:
+                    literal_node = nodes.literal(str(value), str(value))
+                    properties_list += self.make_field(name, literal_node)
+
+                # Special formating of `value` is needed if name == 'choices'
+                elif name == 'choices':
+                    bullet_list = nodes.bullet_list()
+                    for choice in value:
+                        list_item = nodes.list_item()
+                        paragraph = nodes.paragraph()
+                        label = nodes.Text(choice['label'] + ': ')
+                        val = nodes.literal(choice['value'], choice['value'])
+
+                        paragraph += label
+                        paragraph += val
+                        list_item += paragraph
+                        bullet_list += list_item
+
+                    properties_list += self.make_field(name, bullet_list)
+
+                # Make a nested list of properties:
+                elif name == 'group':
+                    list123 = nodes.field_list()
+                    for sub_param in value:
+                        sub_param_properties_list = self.make_properties_list(sub_param)
+                        list123 += self.make_field(sub_param['name'], sub_param_properties_list)
+
+                    properties_list += self.make_field(name, list123)
+
+                else:
+                    properties_list += self.make_field(name, str(value))
+
+        return properties_list
+
+    def make_process_node(self, source_base_url, process):
+        """
+        Fill the content of process definiton node
+
+        :param url/string source_base_url: url to process definition
+        :param dict process: process data as given from yaml.load function
+        :rtype: nodes.field
+        """
+
+        name = process['name']
+
+        # Make process name as section title:
+        section = nodes.section()
+        section['ids'] = [str(uuid.uuid4())]
```

Comment:
Is this id visible to the end user? If yes, it should be nice, i.e. use process slug.